test/vfd_swmr_dsetops_writer.c

```diff
@@ -0,0 +1,1771 @@
+{
+    if (step % s->csteps == 0) {
+        np->verify++;
+        if (HDread(fd, &np->notify, sizeof(int)) < 0) {
+            printf("HDread failed\n");
+            TEST_ERROR;
+        }
+
+        if (np->notify == -1) {
+            printf("reader/writer failed to verify\n");
+            TEST_ERROR;
+        }
+
+        if (np->notify != np->verify) {
+            printf("received message %d, expecting %d\n", np->notify, np->verify);
+            TEST_ERROR;
+        }
+    }
+
+    return true;
+
+error:
+    return false;
+} /* np_confirm_verify_notify() */
+
+/*
+ * Main
+ */
+int
+main(int argc, char **argv)
+{
+    hid_t                 fapl, fcpl;
+    unsigned              step;
+    bool                  writer;
+    state_t               s;
+    const char *          personality;
+    H5F_vfd_swmr_config_t config;
+    np_state_t            np;
+    dsets_state_t         ds;
+    bool                  result;
+
+    if (!state_init(&s, argc, argv)) {
+        printf("state_init() failed\n");
+        TEST_ERROR;
+    }
+
+    personality = HDstrstr(s.progname, "vfd_swmr_dsetops_");
+
+    if (personality != NULL && strcmp(personality, "vfd_swmr_dsetops_writer") == 0)
+        writer = true;
+    else if (personality != NULL && strcmp(personality, "vfd_swmr_dsetops_reader") == 0)
+        writer = false;
+    else {
+        printf("unknown personality, expected vfd_swmr_dsetops_{reader,writer}\n");
+        TEST_ERROR;
+    }
+
+    /* config, tick_len, max_lag, writer, flush_raw_data, md_pages_reserved, md_file_path */
+    init_vfd_swmr_config(&config, 4, 7, writer, FALSE, 128, "./dsetops-shadow");
+
+    /* use_latest_format, use_vfd_swmr, only_meta_page, config */
+    if ((fapl = vfd_swmr_create_fapl(true, s.use_vfd_swmr, true, &config)) < 0) {
+        printf("vfd_swmr_create_fapl() failed\n");
+        TEST_ERROR;
+    }
+
+    if ((fcpl = H5Pcreate(H5P_FILE_CREATE)) < 0) {
+        printf("H5Pcreate failed\n");
+        TEST_ERROR;
+    }
+
+    if (H5Pset_file_space_strategy(fcpl, H5F_FSPACE_STRATEGY_PAGE, false, 1) < 0) {
+        printf("H5Pset_file_space_strategy failed\n");
+        TEST_ERROR;
+    }
+
+    if (writer) {
+        if ((s.file = H5Fcreate(s.filename, H5F_ACC_TRUNC, fcpl, fapl)) < 0) {
+            printf("H5Fcreate failed\n");
+            TEST_ERROR;
+        }
+
+        if (!create_dsets(&s, &ds)) {
+            printf("create_dsets() failed\n");
+            TEST_ERROR;
+        }
+    }
+    else {
+        if ((s.file = H5Fopen(s.filename, H5F_ACC_RDONLY, fapl)) < 0) {
+            printf("H5Fopen failed\n");
+            TEST_ERROR;
+        }
+        if (!open_dsets(&s, &ds)) {
+            printf("open_dsets() failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    /* Initiailze named pipes */
+    if (s.use_np && !np_init(&np, writer)) {
+        printf("np_init() failed\n");
+        TEST_ERROR;
+    }
+
+    if (writer) {
```

Comment:
I feel this part could use some subroutines to simplify the code.

Reply:
Fixed.